library/src/main/java/com/mux/video/upload/internal/TranscoderContext.kt

```diff
@@ -0,0 +1,592 @@
+package com.mux.video.upload.internal
+
+import android.content.Context
+import android.content.ContextWrapper
+import android.media.*
+import android.media.MediaCodec.BufferInfo
+import android.media.MediaCodecInfo.CodecCapabilities.COLOR_FormatYUV420SemiPlanar
+import android.os.Build
+import android.os.Environment
+import android.util.Log
+import io.github.crow_misia.libyuv.FilterMode
+import io.github.crow_misia.libyuv.Nv12Buffer
+import java.io.File
+import java.nio.ByteBuffer
+import java.util.*
+import kotlin.experimental.and
+
+
+class TranscoderContext internal constructor(
+    private val uploadInfo: UploadInfo, private val appContext: Context
+) {
+    val MAX_ALLOWED_BITRATE = 8000000
+    val MAX_ALLOWED_FRAMERATE = 120;
+    val MAX_ALLOWED_WIDTH = 1920
```

Comment:
As part of the API we want to expose a maximum resolution switch so the `MAX_ALLOWED_WIDTH` and `MAX_ALLOWED_HEIGHT can vary. (On iOS the client can pick between 720x1280 and 1920x1080.)